test/helpers/policygen/models.go

```diff
@@ -0,0 +1,664 @@
+	}
+
+	if len(l4egress) > 0 {
+		egressMap[toPorts] = []rule{l4egress}
+	}
+
+	if len(ingressMap) > 0 {
+		specs = append(specs, Spec{
+			EndpointSelector: endpointSelector{
+				Matchlabels: map[string]string{
+					"id": t.DestPod},
+			},
+			Ingress: []rule{ingressMap},
+		})
+	}
+
+	if len(egressMap) > 0 {
+		specs = append(specs, Spec{
+			EndpointSelector: endpointSelector{
+				Matchlabels: map[string]string{
+					"id": t.SrcPod},
+			},
+			Egress: []rule{egressMap},
+		})
+	}
+
+	if len(specs) == 0 {
+		return "", nil
+	}
+
+	jsonOutput, err := json.Marshal(specs)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf(string(metadata), t.Prefix, jsonOutput, t), nil
+}
+
+// NetworkPolicyName returns the name of the NetworkPolicy
+func (t *TestSpec) NetworkPolicyName() string {
+	return fmt.Sprintf("%s_policy.json", t.Prefix)
+}
+
+// NetworkPolicyApply applies the cilium network policy in kubernetes and wait
+// until all  pods updates their status. It'll return an error if pods did not
+// update the state or policy cannot be applied
+func (t *TestSpec) NetworkPolicyApply() error {
+	policy, err := t.NetworkPolicyCreate()
+	if err != nil {
+		return fmt.Errorf("Network policy cannot be created prefix=%s: %s", t.Prefix, err)
+	}
+
+	if policy == "" {
+		//This only happens on L3:No Policy L4:No Policy L7:No Policy
+		log.Info("No policy so do not import it")
+		return nil
+	}
+
+	err = ioutil.WriteFile(t.NetworkPolicyName(), []byte(policy), os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("Network policy cannot be written prefix=%s: %s", t.Prefix, err)
+	}
+
+	_, err = t.Kub.CiliumImportPolicy(
+		helpers.KubeSystemNamespace,
+		fmt.Sprintf("%s/%s", helpers.BasePath, t.NetworkPolicyName()),
+		helpers.HelperTimeout)
+
+	if err != nil {
+		return fmt.Errorf("Network policy cannot be imported prefix=%s: %s", t.Prefix, err)
+	}
+	return nil
+}
+
+type connTestResultType struct {
+	kind   string
+	result ResultType
+}
+
+// getConnectivityTest returns and array with the expected results of the given
+// connectivity test kind
+func (t *TestSpec) getConnectivityTest(kind string) []connTestResultType {
+	return []connTestResultType{
+		connTestResultType{t.l3.kind, t.l3.tests.GetField(kind)},
+		connTestResultType{t.l4.kind, t.l4.tests.GetField(kind)},
+		connTestResultType{t.l7.kind, t.l7.tests.GetField(kind)}}
+}
+
+// GetTestExpects returns a map with the connTestType and the expected result
+// based on the `testExpect`
+func (t *TestSpec) GetTestExpects() map[string]ResultType {
+	expectedTestResult := func(testType string) ResultType {
+		connTest := t.getConnectivityTest(testType)
+
+		//First check the egress rules that are the first rules that match
+		for _, kind := range ConnTestsFailedResults {
+			for _, test := range connTest {
+				if test.kind == egress {
+					if test.result == kind {
+						return kind
+					}
+				}
+			}
+		}
+		// If no resulttype for egress, we need to check if any specific
```

Comment:
Use consistency in comments esp. when referring to types. Change resutltType and resulttype --> ResultType